src/routes/Evodex/BackLayer/Exchange/ExchangeBackLayer.js

```diff
@@ -309,13 +316,43 @@ const ExchangeBackLayer = ({ onReload, ual, isLightMode, showMessage }) => {
     }
 
     const assets = evolutiondex.getExchangeAssets(youGive.inputValue, pair)
-
     setAssets(assets)
     setYouReceive((prevState) => ({
       ...prevState,
       inputValue: assets.assetToReceive.toString().split(' ')[0]
     }))
-  }, [pair, youGive.inputValue])
+  }, [userChangeInput, pair, youGive.inputValue])
+
+  useEffect(() => {
+    if (userChangeInput !== 'youReceive') {
+      return
+    }
+
+    setLastInterval((lastValue) => {
+      clearInterval(lastValue)
+
+      return null
+    })
+
+    if (!pair || !youReceive.inputValue) {
+      setYouGive((prevState) => ({
+        ...prevState,
+        inputValue: ''
+      }))
+      setAssets(null)
+
+      return
+    }
+
+    const assets = evolutiondex.getExchangeAssetsFromToken2(youReceive.inputValue, pair)
+
+    setAssets(assets)
+    setYouGive((prevState) => ({
+      ...prevState,
+      inputValue: assets.assetToGive.toString().split(' ')[0]
+    }))
+    setLastInterval(setTimeout(() => { setUserChangeInput('youGive') }, 2000))
```

Comment:
is 2 seconds the optimal value

Reply:
I am wondering the same thing, maybe @murillojorge could help us with the best timing for the user experience